CMake error when cross-compiling from Linux to Windows

### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

When writing and using a `Windows.cmake` toolchain file to cross-compile from Linux to Windows there's an error:
```cmake
CMake Error at thirdparty/SFML/cmake/Config.cmake:14 (elseif):
  given arguments:

    "" "STREQUAL" "" "AND" "MATCHES" "ARM64"

  Unknown arguments specified
Call Stack (most recent call first):
  thirdparty/SFML/CMakeLists.txt:34 (include)
```

### Your Environment

- OS / distro / window manager: Arch Linux / i3wm
- SFML version: 2.6.2
- Compiler / toolchain: x86_64-w64-mingw32-gcc
- Special compiler / CMake flags:
```cmake
# CMakeLists.txt
cmake_minimum_required(VERSION 3.28)
project(isorotation LANGUAGES CXX)

set(CMAKE_CXX_EXTENSIONS OFF)
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

option(BUILD_SHARED_LIBS "Build shared libraries" OFF)

add_subdirectory(thirdparty/SFML)

add_executable(${PROJECT_NAME} src/main.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE sfml-graphics)

if(WIN32)
    add_custom_command(
        TARGET ${PROJECT_NAME}
        COMMENT "Copy OpenAL DLL"
        PRE_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${SFML_SOURCE_DIR}/extlibs/bin/$<IF:$<EQUAL:${CMAKE_SIZEOF_VOID_P},8>,x64,x86>/openal32.dll $<TARGET_FILE_DIR:main>
        VERBATIM)
endif()

set(source "${CMAKE_SOURCE_DIR}/assets")
set(destination "${CMAKE_CURRENT_BINARY_DIR}/assets")
add_custom_command(
    TARGET ${PROJECT_NAME} POST_BUILD
    COMMAND ${CMAKE_COMMAND} -E create_symlink ${source} ${destination} 
    COMMENT "symbolic link assets folder from ${source} => ${destination}"
)

```
```cmake
# Windows.cmake
set(CMAKE_SYSTEM_NAME Windows)
set(TOOLCHAIN_PREFIX x86_64-w64-mingw32)

# cross compilers to use for C, C++ and Fortran
set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}-gcc)
set(CMAKE_CXX_COMPILER ${TOOLCHAIN_PREFIX}-g++)

# target environment on the build host system
set(CMAKE_FIND_ROOT_PATH /usr/${TOOLCHAIN_PREFIX})

# modify default behavior of FIND_XXX() commands
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)

set(SFML_INCLUDE_DIR "thirdparty/SFML/include")
include_directories(thirdparty/SFML/include)
set(SFML_LIBRARY_DIR "thirdparty/SFML/lib")
set(SFML_DIR "thirdparty/SFML/lib/cmake/SFML")
```
```
cmake -DCMAKE_TOOLCHAIN_FILE=Windows.cmake ..
```

### Steps to reproduce

1. Create any SFML project that links thirdparty/SFML statically.
2. Create a toolchain file for Windows as described above.
3. Create a `build` directory and run `cmake -DCMAKE_TOOLCHAIN_FILE=Windows.cmake ..`

### Expected behavior

To generate make scripts properly and be successful.

### Actual behavior

Error above.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -13,6 +13,7 @@ jobs:
         platform:
         - { name: Windows VS2022,       os: windows-2022 }
         - { name: Windows VS2022 Clang, os: windows-2022, flags: -T ClangCL }
+        - { name: Windows VS2022 arm64, os: windows-2022, flags: -A arm64 -DSFML_BUILD_TEST_SUITE=FALSE }
         - { name: Linux GCC,            os: ubuntu-22.04 }
         - { name: Linux Clang,          os: ubuntu-22.04, flags: -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ }
         - { name: Linux GCC OpenGL ES,  os: ubuntu-22.04, flags: -DSFML_OPENGL_ES=ON }

--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -5,17 +5,12 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
     # don't use the OpenGL ES implementation on Windows
     set(OPENGL_ES 0)
 
-    # detect the architecture (note: this test won't work for cross-compilation)
-    include(CheckTypeSize)
-    check_type_size(void* SIZEOF_VOID_PTR)
-
-    if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
+    # detect the architecture
+    if("${CMAKE_GENERATOR_PLATFORM}" MATCHES "ARM64" OR "${MSVC_CXX_ARCHITECTURE_ID}" MATCHES "ARM64" OR "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "ARM64")
         set(ARCH_ARM64 1)
-    elseif("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "" AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "ARM64")
-        set(ARCH_ARM64 1)
-    elseif(${SIZEOF_VOID_PTR} STREQUAL "4")
+    elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
         set(ARCH_X86 1)
-    elseif(${SIZEOF_VOID_PTR} STREQUAL "8")
+    elseif(CMAKE_SIZEOF_VOID_P EQUAL 8)
         set(ARCH_X64 1)
     else()
         message(FATAL_ERROR "Unsupported architecture")

